### PR TITLE
Fix a11y audit: enlarge touch targets of right side menu "On this page"

### PR DESCRIPTION
**Pull request by hannahiss**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

#2667 issue 1-8

### Description

Increase vertical padding to make the clicking area at least 24px high

### Types of change

- Bug fix (non-breaking which fixes an issue)

### Live previews

- <https://deploy-preview-2726--boosted.netlify.app/docs>

### CODE REVIEW

File: site/assets/scss/_toc.scss [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
     a {
       display: block;
-      padding: .125rem 0 .125rem .75rem;
+      padding: .313rem 0 .25rem .313rem;
```
Comment (julien-deramond) [inline site/assets/scss/_toc.scss:29]: If we modify values from Bootstrap, we should add the "Boosted mod" comment.

```suggestion
      padding: .313rem 0 .25rem .313rem; // Boosted mod
```

Comment (julien-deramond) [reply site/assets/scss/_toc.scss:29]: `.313rem` seems weird to me as a value as it's `5.008px`

**Review by julien-deramond — CHANGES_REQUESTED** [review]
It's no more left-aligned with "On this page" correctly with these values (tablet and desktop breakpoints, mobile seems OK):

Screenshot 2024-09-11 at 08 35 58

Reference on the _main_ branch:

Screenshot 2024-09-11 at 08 36 57

File: site/assets/scss/_toc.scss (hunk 1 above)
Comment (hannahiss) [reply site/assets/scss/_toc.scss]: Sorry I think I was a bit tired last evening 😓

**Review by julien-deramond — APPROVED** [review]

**Pull request merged**